### PR TITLE
[WebSub] Introduce a public function to parse a response received for a discovery request

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
This PR introduces a public function to parse a response received for a discovery request. 

The function would return
- either a tuple (string, string[]) indicating  the topic (self URL) and the hub URL(s)
- or an error if no `Link` header is available, more than one topic (self URL) is specified, no hub/topic is specified, etc.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/10644

### CODE REVIEW

File: stdlib/websub/src/main/ballerina/websub/commons.bal [hunk 1]
```diff
@@ -348,6 +348,58 @@ public type Notification object {
...
+# + response - The `http:Response` received
+# + return - `(topic, hubs)` if parsing and extraction is successful, `error` if not
+public function extractTopicAndHubUrls(http:Response response) returns (string, string[])|error {
+    string[] linkHeaders;
+    error websubError;
+    if (response.hasHeader("Link")) {
+        linkHeaders = response.getHeaders("Link");
+    }
+
+    if (lengthof linkHeaders > 0) {
```
Comment (chamil321) [inline stdlib/websub/src/main/ballerina/websub/commons.bal]: The breath of the code segment can reduce by checking the else part first and return from there.
```
if (lengthof linkHeaders <= 0) {
   return { message: "Link header unavailable in discovery response" };
}
int hubIndex = 0;
string[] hubs;
....

```

File: stdlib/websub/src/main/ballerina/websub/commons.bal [hunk 2]
```diff
@@ -348,6 +348,58 @@ public type Notification object {
...
+                        return websubError;
+                    } else {
+                        topic = url;
+                    }
+                }
+            }
+        }
+        if (lengthof hubs > 0 && topic != "") {
+            return (topic, hubs);
+        } else {
```
Comment (chamil321) [inline stdlib/websub/src/main/ballerina/websub/commons.bal]: You can directly return with error. Else part is not required

File: stdlib/websub/src/main/ballerina/websub/commons.bal [hunk 3]
```diff
@@ -348,6 +348,58 @@ public type Notification object {
...
+            websubError = {message: "Hub and/or Topic URL(s) not identified in link header of discovery response"};
+        }
+    } else {
```
Comment (chamil321) [inline stdlib/websub/src/main/ballerina/websub/commons.bal]: You can directly return with error. Else part is not required

**Comment by MaryamZi** [comment]
@Shan1024, @chamil321, addressed all suggestions, thank you!

**Pull request merged**